core/storage/translation/gae_models.py

```diff
@@ -0,0 +1,147 @@
+        ID.
+
+        Args:
+            source_language_code: str. The language code of the untranslated
+                text.
+            target_language_code: str. The language code of the translation.
+            origin_text: str. The untranslated text.
+            translated_text: str. The translation.
+
+        Returns:
+            str. The id of the newly created MachineTranslatedTextInstance.
+        """
+        hashed_origin_text = utils.convert_to_hash(origin_text, 50)
+        entity_id = cls._generate_id(
+            source_language_code, target_language_code, hashed_origin_text)
+        translation_entity = cls(
+            id=entity_id,
+            hashed_origin_text=hashed_origin_text,
+            source_language_code=source_language_code,
+            target_language_code=target_language_code,
+            origin_text=origin_text,
+            translated_text=translated_text)
+        translation_entity.put()
+        return entity_id
+
+    @classmethod
+    def _generate_id(
+            cls, source_language_code, target_language_code,
+            hashed_origin_text):
+        """Generates a valid key for a MachineTranslatedTextModel.
+
+        Args:
+            source_language_code: str. The language code of the untranslated
+                text.
+            target_language_code: str. The language code of the translation.
+            hashed_origin_text: str. An sha-1 hash of the origin_text.
+
+        Returns:
+            str. The generated ID for this entity of the form
+
+            [source_language_code]:[target_language_code]:[hashed_origin_text].
+        """
+
+        return (
+            '%s:%s:%s' % (
+                source_language_code, target_language_code, hashed_origin_text)
+        )
+
+    @classmethod
+    def get_translation_for_text(
+            cls, source_language_code, target_language_code, origin_text):
+        """Gets MachineTranslatedTextModel by language codes and origin text.
+        Returns None if no translation exists for the given parameters.
+
+        Args:
+            source_language_code: str. The language of the origin_text.
+            target_language_code: str. The language being translated to.
+            origin_text: str. The text to be translated.
+
+        Returns:
+            MachineTranslatedTextModel|None. The MachineTranslatedTextModel
+            if a translation exists or None if no translation is found.
+        """
+        hashed_origin_text = utils.convert_to_hash(origin_text, 50)
+        instance_id = cls._generate_id(
+            source_language_code, target_language_code, hashed_origin_text)
+        return cls.get(instance_id)
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model doesn't contain any data directly corresponding to a user."""
+        return base_models.DELETION_POLICY.NOT_APPLICABLE
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model doesn't contain user data."""
```

Comment:
```suggestion
        """Model is not associated with users."""
```

I think this might be a bit clearer.

Reply:
Fixed.